low-level/sensors/impl/generic/node_hw.py

```diff
@@ -480,6 +481,58 @@ def run(self):
             logger.warning(f"{self.SENSOR_NAME} Node hw monitoring disabled")
             self.shutdown()
 
+    def _check_missmatch_resource_status(self):
+        for sensor_id in self.faulty_resources:
+            if self.faulty_resources[sensor_id]['resource'] in self.fru_types.keys():
+                if self.faulty_resources[sensor_id]['resource'] in [
+                    self.TYPE_PSU_SUPPLY, self.TYPE_PSU_UNIT]:
+                    sdr_get_data, _ = self._get_sensor_sdr_props(sensor_id)
+                    if sdr_get_data and 'States Asserted' in sdr_get_data:
+                        #  'States Asserted': 'Power Supply, Presence detected'
+                        resource_state = re.sub(',  +', ', ', re.sub('[\[\]]','',
+                            sdr_get_data['States Asserted']).replace('\n',','))
+                        sensor_status = resource_state.split(',')
+                        sensor_status = sensor_status[0] if len(sensor_status) == 1 \
+                            else sensor_status[1]
+                        sensor_status = sensor_status.strip()
+                        if sensor_status in ["", "Presence detected"]:
+                            self._create_sel_event_entry(sensor_id)
+                else:
+                    _, sensor_get_data, _ = self._get_sensor_props(sensor_id)
+                    if sensor_get_data and 'Assertion Events' in sensor_get_data \
+                            and 'Status' in sensor_get_data:
+                        # 'Status': 'ok'
+                        # 'Assertion Events': ''
+                        if sensor_get_data['Assertion Events'] in [""] \
+                            and sensor_get_data['Status'] == 'ok':
+                            self._create_sel_event_entry(sensor_id)
+
+    def _create_sel_event_entry(self, sensor_id):
+        event_map_dict = {
+            'lower non-recoverable': 'lnr',
+            'lower critical': 'lcr',
+            'lower non-critical': 'lnc',
+            'upper non-critical': 'unc',
+            'upper critical': 'ucr',
+            'upper non-recoverable': 'unr'
+        }
+        event = self.faulty_resources[sensor_id]['event']
+        if event.lower().endswith(("going high", "going low")):
+            event = next(v for k,v in event_map_dict.items() \
+                if event.lower().startswith(k))
+        # "Presence detected ()"
+        event = event.replace('()', '').strip()
+        status = self.faulty_resources[sensor_id]['status'].lower()
+        status = 'deasserted' if status == 'asserted' else 'asserted'
+        logger.warn("Found missing entry in the ipmitool sel list for %s resource,\
+            Updating the sel list manually" % (sensor_id))
+        # ipmitool event 'PS1 Status' 'Presence detected' deasserted
+        event_cmd = "event '%s' '%s' %s" % (sensor_id, event, status)
```

Comment:
event command is to trigger a test event, not supposed to be used to for any user generated event, check its man page

>event
\<predefined event number N\>
Send a pre-defined test event to the System Event Log. The following events are included as a means to test the functionality of the System Event Log component of the BMC (an entry will be added each time the event N command is executed).

>Currently supported values for N are:
1 Temperature: Upper Critical: Going High
2 Voltage Threshold: Lower Critical: Going Low
3 Memory: Correctable ECC

>NOTE: These pre-defined events will likely not produce "accurate" SEL records for a particular system because they will not be correctly tied to a valid sensor number, but they are sufficient to verify correct operation of the SEL.

Reply:
Done